src/hypothesis/strategies.py

```diff
@@ -910,6 +912,110 @@ def do_draw(self, data):
+
+    If ``timezones`` is not None, it must be a strategy that generates tzinfo
+    objects (or None, which is valid for naive datetimes).  A value drawn
+    from this strategy will be added to a naive datetime, and the resulting
+    tz-aware datetime returned.
+
+    .. note::
+        tz-aware datetimes from this strategy may be ambiguous or non-existent
+        due to daylight savings, leap seconds, timezone and calendar
+        adjustments, etc.  This is intentional, as malformed timestamps are a
+        common source of bugs.  Consider validating all datetime values, or
+        working purely in UTC to avoid such issues.
+
+    There are a number of ways you can create a timezones strategy, depending
+    mostly on your timezone objects (e.g. from the standard library,
+    ``datetutil``, or ``pytz``).  In each case, you should probably use
+    :py:func:`sampled_from` to sample from a sequence.  To minimise
+    effectively, consider composing strategies for None, UTC, and other
+    timezones in that order (e.g. using :py:func:`one_of`).
+
+    :py:func:`hypothesis.extra.timezones` requires the ``pytz`` package, but
+    provides all timezones in the Olsen database.  If you also want to allow
+    naive datetimes, combine strategies like ``none() | timezones()``.
+
+    """
+    from hypothesis.searchstrategy.datetime import DatetimeStrategy
+
+    check_type(dt.datetime, min_datetime, 'min_datetime')
+    check_type(dt.datetime, max_datetime, 'max_datetime')
+    if min_datetime.tzinfo is not None:
+        raise InvalidArgument('min_datetime=%r must not have tzinfo'
+                              % min_datetime)
+    if max_datetime.tzinfo is not None:
+        raise InvalidArgument('max_datetime=%r must not have tzinfo'
+                              % max_datetime)
+    check_valid_interval(min_datetime, max_datetime,
+                         'min_datetime', 'max_datetime')
+    if timezones is None:
+        if min_datetime == max_datetime:
+            return just(min_datetime)
+        timezones = none()
+    if not isinstance(timezones, SearchStrategy):
+        raise InvalidArgument(
+            'timezones=%r must be a SearchStrategy that can provide tzinfo '
+            'for datetimes (either None or dt.tzinfo objects)' % (timezones,))
+    return DatetimeStrategy(min_datetime, max_datetime, timezones)
+
+
+@defines_strategy
+def dates(min_date=dt.date.min, max_date=dt.date.max):
+    """A strategy for dates between ``min_date`` and ``max_date``."""
+    check_type(dt.date, min_date, 'min_date')
+    check_type(dt.date, max_date, 'max_date')
+    check_valid_interval(min_date, max_date, 'min_date', 'max_date')
+    if min_date == max_date:
+        return just(min_date)
+    return datetimes(min_datetime=dt.datetime.combine(min_date, dt.time.min),
+                     max_datetime=dt.datetime.combine(max_date, dt.time.max)
+                     ).map(dt.datetime.date)
```

Comment:
Those scenarios are only representable with tz-aware datetimes, luckily.  I agree that the delta-based strategy is nicer, but the advantage of using the datetimes strat is it minimises towards 2000-01-01 rather than min_date (usually 0001-01-01).